pkg/asset/manifests/powervs/cluster.go

```diff
@@ -0,0 +1,202 @@
+			Name: &serviceName,
+		}
+	} else {
+		service = capibm.IBMPowerVSResourceReference{
+			ID: &installConfig.Config.PowerVS.ServiceInstanceGUID,
+		}
+	}
+
+	vpcName = installConfig.Config.Platform.PowerVS.VPCName
+	if vpcName == "" {
+		vpcName = fmt.Sprintf("vpc-%s", clusterID.InfraID)
+	}
+
+	vpcRegion = installConfig.Config.Platform.PowerVS.VPCRegion
+	if vpcRegion == "" {
+		if vpcRegion, err = powervstypes.VPCRegionForPowerVSRegion(installConfig.Config.PowerVS.Region); err != nil {
+			return nil, fmt.Errorf("unable to derive vpcRegion from region: %s %w", installConfig.Config.PowerVS.Region, err)
+		}
+	}
+
+	transitGatewayName = fmt.Sprintf("%s-tg", clusterID.InfraID)
+
+	cosName = fmt.Sprintf("%s-cos", clusterID.InfraID)
+
+	imageName = fmt.Sprintf("rhcos-%s", clusterID.InfraID)
+
+	bucketName = fmt.Sprintf("%s-bootstrap-ign", clusterID.InfraID)
+
+	powerVSCluster = &capibm.IBMPowerVSCluster{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: capibm.GroupVersion.String(),
+			Kind:       "IBMPowerVSCluster",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterID.InfraID,
+			Namespace: capiutils.Namespace,
+			Annotations: map[string]string{
+				"powervs.cluster.x-k8s.io/create-infra": "true",
+			},
+		},
+		Spec: capibm.IBMPowerVSClusterSpec{
+			Network: capibm.IBMPowerVSResourceReference{
+				Name: &network,
+			},
+			ServiceInstance: &service,
+			Zone:            &installConfig.Config.Platform.PowerVS.Zone,
+			ResourceGroup: &capibm.IBMPowerVSResourceReference{
+				Name: &installConfig.Config.Platform.PowerVS.PowerVSResourceGroup,
+			},
+			VPC: &capibm.VPCResourceReference{
+				Name:   &vpcName,
+				Region: &vpcRegion,
+			},
+			TransitGateway: &capibm.TransitGateway{
+				Name: &transitGatewayName,
+			},
+			LoadBalancers: []capibm.VPCLoadBalancerSpec{
+				{
+					Name:   fmt.Sprintf("%s-loadbalancer", clusterID.InfraID),
+					Public: ptr.To(true),
+					AdditionalListeners: []capibm.AdditionalListenerSpec{
+						{
+							Port: 22,
+						},
+						// @BUG We should be able to specify this:
+						// capibm.AdditionalListenerSpec{
+						//	Port: 6443,
+						// },
+					},
+				},
+				{
+					Name:   fmt.Sprintf("%s-loadbalancer-int", clusterID.InfraID),
+					Public: ptr.To(false),
+					AdditionalListeners: []capibm.AdditionalListenerSpec{
+						// @BUG We should be able to specify this:
+						// capibm.AdditionalListenerSpec{
+						//	Port: 6443,
+						// },
+						{
+							Port: 22623,
+						},
+					},
+				},
+			},
+			CosInstance: &capibm.CosInstance{
+				Name:         cosName,
+				BucketName:   bucketName,
+				BucketRegion: vpcRegion,
+			},
+			Ignition: &capibm.Ignition{
+				Version: "3.4",
+			},
+		},
+	}
+
+	// Avoid empty manifest elements
+	logrus.Debugf("GenerateClusterAssets: len(VPCSubnets) = %d", len(installConfig.Config.Platform.PowerVS.VPCSubnets))
+	if len(installConfig.Config.Platform.PowerVS.VPCSubnets) > 0 {
+		powerVSCluster.Spec.VPCSubnets = make([]capibm.Subnet, 0)
```

Comment:
Nit: can preallocate the array here
```suggestion
		powerVSCluster.Spec.VPCSubnets = make([]capibm.Subnet, 0, len(installConfig.Config.Platform.PowerVS.VPCSubnets)
```